src/vanilla.ts

```diff
@@ -4,8 +4,8 @@ export type PartialState<T extends State> =
   | ((state: T) => Partial<T>)
 export type StateSelector<T extends State, U> = (state: T) => U
 export type EqualityChecker<T> = (state: T, newState: unknown) => boolean
-export type StateListener<T> = (state: T) => void
-export type StateSliceListener<T> = (state: T | null, error?: Error) => void
+export type StateListener<T> = (state: T, previousState: T) => void
+export type StateSliceListener<T> = (state: T, previousState: T) => void
```

Comment:
```suggestion
export type StateSliceListener<T> = (slice: T, previousSlice: T) => void
```

Let's just rename it.